helix-term/src/application.rs

```diff
@@ -519,6 +516,13 @@ impl Application {
                                 .collect();
 
                             doc.set_diagnostics(diagnostics);
+
+                            // Insert the original lsp::Diagnostics here because we may have no open document
+                            // for diagnosic message and so we can't calculate the exact position.
+                            // When using them later in the diagnostics picker, we calculate them on-demand.
+                            self.editor
+                                .diagnostics
+                                .insert(params.uri, params.diagnostics);
                         }
```

Comment:
```suggestion
                        }
                        // Insert the original lsp::Diagnostics here because we may have no open document
                        // for diagnosic message and so we can't calculate the exact position.
                        // When using them later in the diagnostics picker, we calculate them on-demand.
                        self.editor
                            .diagnostics
                            .insert(params.uri, params.diagnostics);
```

Reply:
The issue appears to have been that `if let Some(doc) = doc` doesn't work when the file is not opened, so moving the insert outside the if makes all diagnostics show for me.

Reply:
Yes you're right. This bug was added with my last commit. Thank you.